### PR TITLE
Add Hebrew translation and RTL support (fixes #372)

### LINKED ISSUE #372: RTL support
#### I'm submitting a ... 
  - [ ] bug report
  - [x] feature request
 
#### What is the current behavior?

Currently Right-To-Left languages is not supported by Allure. 

#### What is the expected behavior?

Allure supports Right-To-Left languages.

#### What is the motivation / use case for changing the behavior?

I want to add Hebrew translation to Allure.

#### Please tell us about your environment:

- Allure version: any

#### Other information 

We can use [postcss-rtl](https://github.com/vkalinichev/postcss-rtl) module for base support for RTL languages.

### PR DESCRIPTION
**Pull request by TpSr52**
It's my first contribution to an open-source project! 
Allure 2 doesn't have RTL support (fix #372\) and Hebrew translation

Fix:
- I have added postcss-rtl as a dev dependency in package.json and in webpack.config.js in order to support RTL.
- I have added the file rtl.css and I changed some css files in order to fix some RTL related issues.
- I have added a translation to Hebrew.

### CODE REVIEW

**Comment by TpSr52** [comment]
Split.js has a bug with flexbox when the page is in RTL. I think about a solution and I have submitted an issue in their Github. I hope there will be any fix for it soon.

**Comment by TpSr52** [comment]
Does anyone want to review?  :)

File: allure-generator/src/main/javascript/app.js [hunk 1]
```diff
@@ -46,9 +46,11 @@ const App = new Application({
 App.on('start', () => {
     initTranslations().then(() => {
         history.start();
+		document.dir = i18next.dir();
         i18next.on('languageChanged', () => {
             App.getRegion().reset();
             router.reload();
+			document.dir = i18next.dir();
```
Comment (baev) [inline allure-generator/src/main/javascript/app.js]: indent

File: allure-generator/src/main/javascript/app.js [hunk 2]
```diff
@@ -46,9 +46,11 @@ const App = new Application({
 App.on('start', () => {
     initTranslations().then(() => {
         history.start();
+		document.dir = i18next.dir();
```
Comment (baev) [inline allure-generator/src/main/javascript/app.js]: indent

File: allure-generator/package.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
   "author": "Dmitry Baev",
   "license": "Apache-2.0",
   "devDependencies": {
+	"postcss-rtl": "^0.5.10",
```
Comment (baev) [inline allure-generator/package.json]: indent

File: allure-generator/src/main/javascript/components/graph-trend-chart/TrendChartView.js [hunk 4]
```diff
@@ -68,10 +68,19 @@ class TrendChartView extends BaseChartView {
...
+		
+        if(document.dir === 'rtl'){
+            this.svg.selectAll('.chart__axis_x')
+            .selectAll('text')
+            .style('text-anchor', 'start');
+		} else {
+			this.svg.selectAll('.chart__axis_x')
+            .selectAll('text')
+            .style('text-anchor', 'end');
+		}
```
Comment (baev) [inline allure-generator/src/main/javascript/components/graph-trend-chart/TrendChartView.js]: indent

File: allure-generator/src/main/javascript/components/graph-trend-chart/TrendChartView.js [hunk 5]
```diff
@@ -68,10 +68,19 @@ class TrendChartView extends BaseChartView {
         this.makeLeftAxis({
             scale: this.y
         });
+
```
Comment (baev) [inline allure-generator/src/main/javascript/components/graph-trend-chart/TrendChartView.js]: remove tab

File: allure-generator/src/main/javascript/components/tooltip/TooltipView.js [hunk 6]
```diff
@@ -71,11 +71,22 @@ class TooltipView extends View {
...
+			this.$el.css(POSITION['right'](
+            anchor[0].getBoundingClientRect(),
+            {offset: this.options.offset},
+            this.$el[0].getBoundingClientRect()));
+		} else {
+			this.$el.css(POSITION[position](
+            anchor[0].getBoundingClientRect(),
+            {offset: this.options.offset},
+            this.$el[0].getBoundingClientRect()));
+		}
```
Comment (baev) [inline allure-generator/src/main/javascript/components/tooltip/TooltipView.js]: indents

File: allure-generator/src/main/javascript/utils/translation.js [hunk 7]
```diff
@@ -5,7 +5,8 @@ export const LANGUAGES = [
     {id: 'en', title: 'English'},
     {id: 'ru', title: 'Русский'},
     {id: 'zh', title: '中文'},
-    {id: 'de', title: 'Deutsch'}
+    {id: 'de', title: 'Deutsch'},
+	{id: 'he', title: 'Hebrew'}
```
Comment (baev) [inline allure-generator/src/main/javascript/utils/translation.js]: indent

File: allure-generator/src/main/javascript/components/side-nav/styles.scss [hunk 8]
```diff
@@ -44,7 +44,7 @@
 
   &__footer {
     margin-top: auto;
-    margin-bottom: $gap-size;
+    margin-bottom: $footer-gap-size;
```
Comment (baev) [inline allure-generator/src/main/javascript/components/side-nav/styles.scss]: what is the point of this change?

File: allure-generator/webpack.config.js [hunk 9]
```diff
@@ -7,7 +7,8 @@ const postcssLoader = {
     loader: 'postcss-loader',
     options: {
         plugins: [
-            require('autoprefixer')
+            require('autoprefixer'),
+			require('postcss-rtl')
```
Comment (baev) [inline allure-generator/webpack.config.js]: indent

File: allure-generator/src/main/javascript/components/side-nav/styles.scss (hunk 8 above)
Comment (TpSr52) [inline allure-generator/src/main/javascript/components/side-nav/styles.scss]: I have added another gap-size const variable in order to lift the languages bar a little upper.
When I added Hebrew and opened the language bar, the Hebrew language was sliced by the page (overflow).

**Comment by baev** [comment]
@TpSr52 please sign our CLA https://cla-assistant.io/allure-framework/allure2?pullRequest=475

File: allure-generator/src/main/javascript/components/side-nav/styles.scss (hunk 8 above)
Comment (baev) [inline allure-generator/src/main/javascript/components/side-nav/styles.scss]: > When I added Hebrew and opened the language bar, the Hebrew language was sliced by the page (overflow).

@just-boris is there a way to fix it in simple way? I guess we can change position from `right` to `top-right`

screen shot 2017-08-20 at 20 32 12

**Comment by baev** [comment]
@TpSr52 thanks!

**Pull request merged**